server/@types/forms/index.d.ts

```diff
@@ -34,6 +34,7 @@ declare module 'forms' {
     note?: string
     steps: Array<UpdateStepForm>
     action?: 'add-another-step' | 'submit-form' | 'delete-step-[0]'
+    originalTargetCompletionDate: string
```

Comment:
We now hold the original target completion date (from the original un-edited Goal) in the form, so that we know what it was (in the case of the user entering an invalid date)